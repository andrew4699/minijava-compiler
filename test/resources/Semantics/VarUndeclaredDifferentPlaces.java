class Foo {

  public static void main(String[] args) {
    System.out.println(new Bar().buzz());
  }
}

class Bar {
  int k; int m;
  public int buzz() {
    x = 10;
    return 5;
  }

  public boolean bazz() {
    x = 11;
    return 6;
  }
}

class Buzz {
  int k;
  public int buzz() {
    x = 10;
    return 5;
  }
}