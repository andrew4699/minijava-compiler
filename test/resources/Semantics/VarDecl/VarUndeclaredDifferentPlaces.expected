[VarDeclVisitor - 11]: x undeclared at Bar.buzz
[VarDeclVisitor - 11]: x undeclared at Bar.buzz
[VarDeclVisitor - 16]: x undeclared at Bar.bazz
[VarDeclVisitor - 16]: x undeclared at Bar.bazz
[VarDeclVisitor - 24]: x undeclared at Buzz.buzz
[VarDeclVisitor - 24]: x undeclared at Buzz.buzz
